Firmware/Marlin_main.cpp

```diff
@@ -6512,6 +6513,39 @@ SERIAL_PROTOCOLPGM("\n\n");
     }
     break;
 
+    /*!
+    ### M118 - Serial print <a href="https://reprap.org/wiki/G-code#M118:_Echo_message_on_host">M118: Serial print</a>
+    #### Usage
+
+        M118 [ A1 | E1 ] [ String ]
+
+    #### Parameters
+    - `A1` - Prepend // to denote a comment or action command. Hosts like OctoPrint can interpret such commands to perform special actions. See your host’s documentation.
+    - `E1` - Prepend echo: to the message. Some hosts will display echo messages differently when preceded by echo:.
+    - `String` - Message string. If omitted, a blank line will be sent.
+    */
+    case 118: {
+        bool hasE, hasA = false;
+        char *p = strchr_pointer;
+        
+        for (uint8_t i = 2; i--;) {
+          // A1, E1, and Pn are always parsed out
+          if (!((p[0] == 'A' || p[0] == 'E') && p[1] == '1')) break;
```

Comment:
This line assumes there are always 2 characters in the input buffer (well, the memory of the buffer is valid all the time, but I'm not sure if it gets cleared after each G-code). Can we presume safely, that the code survives malformed syntax like `M118 A`?

Reply:
To be honest I don't really like how `A1` and `E1` are used instead of just plain old `A` and `E`, but just to adhere to the protocol that's already out there I would propose sticking to it.
Similarly we should stick to the same method of parsing it, such that the results replicate between stock marlin and Prusa FW. If the reference just prints the parameter flags when the command is malformed so should we.

As it stands, whenever the command flags are malformed, the entire string including the flags is just echoed over serial (depending on what parts of the command are malformed). Based on some preliminary testing the execution of this code should be safe.

I've got a code snipped in [Godbolt.org](https://godbolt.org/z/E859fTTsc) to validate that.